app/frontend/src/features/communities/CommunityPage.tsx

```diff
@@ -0,0 +1,270 @@
+import PageTitle from "../../components/PageTitle";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Breadcrumbs } from "@material-ui/core";
+import { Link } from "react-router-dom";
+import { Community } from "../../pb/communities_pb"
+import { service } from "../../service";
+import Alert from "../../components/Alert";
+import CircularProgress from "../../components/CircularProgress";
+import TextBody from "../../components/TextBody";
+import { useHistory } from "react-router-dom";
+import { communityRoute, groupRoute, pageRoute } from "../../AppRoutes"
+import Markdown from "../../components/Markdown";
+import { Group } from "../../pb/groups_pb";
+import { Page } from "../../pb/pages_pb";
+
+export default function CommunityPage() {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [community, setCommunity] = useState<Community.AsObject | null>(null);
```

Comment:
These should probably `useSafeState` with `useIsMounted` as it's possible to navigate away during the async call, so they might try to set state on an unmounted component. Same for any state which is changed in an async function.

Reply:
Maybe you don't need to bother though, this will all be replaced with react-query hooks.